ui/src/trace_processor/http_rpc_engine.ts

```diff
@@ -86,11 +88,30 @@ export class HttpRpcEngine extends EngineBase {
   }
 
   private onWebsocketMessage(e: MessageEvent) {
-    assertExists(e.data as Blob)
-      .arrayBuffer()
-      .then((buf) => {
+    const blob = assertExists(e.data as Blob);
+    this.queue.push(blob);
+    this.processQueue();
+  }
+
+  private async processQueue() {
+    if (this.isProcessingQueue) return;
+    this.isProcessingQueue = true;
+    try {
+      while (this.queue.length > 0) {
+        // retain the previous behavior which use the global
+        // reportError to display error messages.
+        const blob = assertExists(this.queue.shift());
+        const buf = await blob.arrayBuffer();
         super.onRpcResponseBytes(new Uint8Array(buf));
-      });
+      }
+    } finally {
+      this.isProcessingQueue = false;
+      // if the queue is not empty, we need start a new task
+      // to continue process the remain websocket message.
+      if (this.queue.length > 0) {
+        this.processQueue();
+      }
```

Comment:
isn't this what the while does? can you explain in which case this is needed?
of is it for the exception case?

Reply:
I think the try..catch could be moved inside the while loop and we could omit this bit